CryptoTechReminderSystem.Main/Program.cs

```diff
@@ -1,43 +1,97 @@
+        public ReminderRegistry()
+        {
+            var slackGateway = new SlackGateway(
+                "https://slack.com/",
+                Environment.GetEnvironmentVariable("SLACK_TOKEN")
+            );
+            var harvestGateway = new HarvestGateway(
+                "https://api.harvestapp.com/",
+                Environment.GetEnvironmentVariable("HARVEST_TOKEN"),
+                Environment.GetEnvironmentVariable("HARVEST_ACCOUNT_ID")
             );
-            
-            IList<string> idsList = new List<string>()
-            {
-                "UH3DVNTH7", 
-                "UH9NYEWSC", 
-                "UHC4PEXM2", 
-                "UHCABC886", 
-                "UHCFC7Z61", 
-                "UHDJJA52S"
-            };
+            _clock = new Clock();
+            _getLateDevelopers = new GetLateDevelopers(slackGateway, harvestGateway, harvestGateway, _clock);
+            _sendReminder = new SendReminder(slackGateway);
 
-            foreach (var id in idsList)
-            {
-                remindDeveloper.Execute(new SendReminderRequest
+            CreateSchedule();
+        }
+
+        private void ResetSchedule()
+        {
+            // Stops scheduled jobs then sets them up again for next week
+            JobManager.RemoveAllJobs();
+            CreateSchedule();
+        }
+        
+        private void CreateSchedule()
+        {
+            // Creates schedule that starts and stops schedule jobs
+            JobManager.AddJob(ScheduleJobs, s => s.ToRunEvery(1).Weeks().On(DayOfWeek.Friday).At(10,30));
+            JobManager.AddJob(ResetSchedule, s => s.ToRunEvery(1).Weeks().On(DayOfWeek.Friday).At(13,45));
+        }
+
+        private void ScheduleJobs()
+        {
+            // Do all of your actual job scheduling here
+            JobManager.AddJob(RemindLateDevelopersJob, s => s.ToRunEvery(30).Minutes());
+            JobManager.AddJob(ShameLateDevelopersJob, s => s.ToRunOnceAt(13, 30));
+        }
+        
+        private void RemindLateDevelopersJob()
+        {
+            var remindLateDevelopers = new RemindLateDevelopers(_getLateDevelopers, _sendReminder, _clock);
+            remindLateDevelopers.Execute(
+                new RemindLateDevelopersRequest
                 {
-                    Channel = id,
-                    Text = "Please make sure your timesheet is submitted by 13:30 on Friday."
-                });
-            }
+                    Message = "Please make sure your timesheet is submitted by 13:30 on Friday."
+                }
+            );
+        }
+        
+        private void ShameLateDevelopersJob()
+        {
+            var shameLateDevelopers = new ShameLateDevelopers(_getLateDevelopers, _sendReminder, _clock);
+            shameLateDevelopers.Execute(
+                new ShameLateDevelopersRequest
+                {
+                    Message = "These are the people yet to submit timesheets:",
+                    Channel = "CGHBW6YGM"
```

Comment:
How do we know what channel this is? Could this be an environment variable as well?

Reply:
👍

Reply:
Resolved in https://github.com/chubberlisk/crypto_tech/pull/25/commits/9345d3c8934d428d33b5c6be1f01accb018def22.